### PR TITLE
Crpl 1518 data/array nodes help

### PR DESCRIPTION
**Pull request by csabee1979**

### CODE REVIEW

File: data/array/filter/a3f6c365-be00-4e30-9772-877f6c38c4f3.json [hunk 1]
```diff
@@ -1,27 +1,27 @@
 {
-  "description": "Filters array. External node(s) receive each item and decide whether to include them in the result.",
+  "description": "Filters array. External node(s) receive each item and decide whether to include them in the result.\n\nExample:\nhttps://github.com/Cranq-io/cranq-tutorials/blob/main/reference/3_querying_data/3_2_filters/README.md#using-filters",
```
Comment (lvass74) [inline data/array/filter/a3f6c365-be00-4e30-9772-877f6c38c4f3.json:2]: "Filters **the** array" or "Filters **an** array" or "Filters `array`" ?
For me something is missing here.

File: data/array/filter/a3f6c365-be00-4e30-9772-877f6c38c4f3.json [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
   "interface": {
     "inputs": {
       "07c8d06d-f187-4600-8bfe-7ab740adebb8": {
-        "description": "Array to be filtered",
+        "description": "Receives array to be filtered.",
```
Comment (lvass74) [inline data/array/filter/a3f6c365-be00-4e30-9772-877f6c38c4f3.json:6]: Receives **the** array...

File: data/array/filter/a3f6c365-be00-4e30-9772-877f6c38c4f3.json [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
         "type": "any[]",
         "name": "array"
       },
       "ff3a8fe6-f8de-43df-ae32-fa0a27bb6ba3": {
-        "description": "Whether to include the current (by tag) item.",
+        "description": "Receives whether to include the current (by tag) item.",
```
Comment (lvass74) [inline data/array/filter/a3f6c365-be00-4e30-9772-877f6c38c4f3.json:11]: Not sure if I'd understand what "(by tag)" means here

File: data/array/first item getter/5bead7ea-5031-4993-a855-dc9ca089802f.json [hunk 4]
```diff
@@ -1,17 +1,17 @@
 {
-  "description": "Retrieves first item of the array.",
+  "description": "Retrieves first item of the array.\n\nExample:\n1. [1,2]@0 received via `array`\n2. 1@0 sent via `item`\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/blob/main/reference/3_querying_data/3_1_getters/README.md#getting-the-first--last-array-elements",
   "interface": {
     "inputs": {
       "cfd3b3e7-80fb-464a-a923-f9fcb9555b0c": {
-        "description": "Array to retrieve first item from",
+        "description": "Receives array to retrieve first item from.",
```
Comment (lvass74) [inline data/array/first item getter/5bead7ea-5031-4993-a855-dc9ca089802f.json:6]: Receives **the** array...

File: data/array/first item getter/5bead7ea-5031-4993-a855-dc9ca089802f.json [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
         "type": "any[]",
         "name": "array"
       }
     },
     "outputs": {
       "f7832747-fb1e-432b-9c10-60ac76c110cc": {
-        "description": "First item of the array",
-        "type": "typeof(array)[number]",
+        "description": "Sends first item of the array.",
```
Comment (lvass74) [inline data/array/first item getter/5bead7ea-5031-4993-a855-dc9ca089802f.json:13]: Sends **the** first item...

File: data/array/flattener/1833c846-f3b9-430a-9b20-c564d5f65c3f.json [hunk 6]
```diff
@@ -65,19 +65,25 @@
   "interface": {
     "inputs": {
       "67681054-c95b-4da1-90d6-1a5461612c4c": {
-        "type": "any",
-        "name": "2D array"
+        "type": "any[][]",
+        "name": "2D array",
+        "description": "Recieves 2D arrays to be flattened."
```
Comment (lvass74) [inline data/array/flattener/1833c846-f3b9-430a-9b20-c564d5f65c3f.json:70]: Receives **a 2D array** to be flattened.

File: data/array/flattener/1833c846-f3b9-430a-9b20-c564d5f65c3f.json [hunk 7]
```diff
@@ -65,19 +65,25 @@
...
+        "name": "array",
+        "description": "Sends flattened array."
       }
     }
   },
   "name": "data/array/Flattener",
   "attributes": {
     "locked": true
+  },
+  "description": "Flattens a 2D array into a  1 D array with.\n\nExample:\n1. [[1,2], [3,4]]@0 recieved via `2D array`\n2.  [1,2,3,4]@0 sent via `array` ",
```
Comment (lvass74) [inline data/array/flattener/1833c846-f3b9-430a-9b20-c564d5f65c3f.json:85]: Some extra spaces in "into a 1D array"

File: data/array/has same length tester/8eeaa000-c916-4045-808d-b918f24d1aa2.json [hunk 8]
```diff
@@ -94,23 +94,30 @@
...
+        "description": "Recieves array2 to compare its lenght."
       }
     },
     "outputs": {
       "dac25b79-8a78-457d-9c9c-52cb867f85a0": {
-        "type": "any",
-        "name": "has same length"
+        "type": "boolean",
+        "name": "has same length",
+        "description": "Sends whether length of array1 and array1 is the same."
```
Comment (lvass74) [inline data/array/has same length tester/8eeaa000-c916-4045-808d-b918f24d1aa2.json:111]: array1 and array2 should be wrapped in backticks

File: data/array/item appender/9882ea6b-43da-4a1f-b660-689da949c8f2.json [hunk 9]
```diff
@@ -58,23 +58,30 @@
   "interface": {
     "inputs": {
       "5cbbb232-d9b4-4d06-9db0-e548e44e43b6": {
-        "type": "any",
-        "name": "array"
+        "type": "any[]",
+        "name": "array",
+        "description": "Receives array to append item to."
```
Comment (lvass74) [inline data/array/item appender/9882ea6b-43da-4a1f-b660-689da949c8f2.json:63]: Receives **the** array

File: data/array/item appender/9882ea6b-43da-4a1f-b660-689da949c8f2.json [hunk 10]
```diff
@@ -58,23 +58,30 @@
...
       },
       "2d02022b-c114-4207-935e-11490a6c2926": {
         "type": "any",
-        "name": "item"
+        "name": "item",
+        "description": "Receives item to append to the array."
```
Comment (lvass74) [inline data/array/item appender/9882ea6b-43da-4a1f-b660-689da949c8f2.json:68]: Receives **the** item

File: data/array/item appender/9882ea6b-43da-4a1f-b660-689da949c8f2.json [hunk 11]
```diff
@@ -58,23 +58,30 @@
...
       }
     }
   },
   "name": "data/array/Item appender",
   "attributes": {
     "locked": true
-  }
+  },
+  "keywords": {
+    "add": null
```
Comment (lvass74) [inline data/array/item appender/9882ea6b-43da-4a1f-b660-689da949c8f2.json:84]: Maybe add "push" as in JS?

File: data/array/item appender/internal/9efd883d-58f1-4091-8d4e-1b8585feaf37.json [hunk 12]
```diff
@@ -9,13 +9,13 @@
...
       "313b183a-6816-45e9-abb7-dd5319a48fbd": {
-        "type": "any",
+        "type": "{\"array\": any[], \"item\": any}",
         "name": "array & item"
       }
     },
     "outputs": {
       "5de10da8-d82e-4757-9e51-fd1815c2e57b": {
-        "type": "any",
+        "type": "(typeof `array & item`)[\"array\"]",
```
Comment (lvass74) [inline data/array/item appender/internal/9efd883d-58f1-4091-8d4e-1b8585feaf37.json:18]: Shouldn't it be just ```typeof `array` ``` as item should be the same type as the items of array?

File: data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json [hunk 13]
```diff
@@ -1,22 +1,22 @@
 {
-  "description": "Deletes item from the array at the specified index.",
+  "description": "Deletes item from the array at the specified index.\n\nExample: \n1.  [1,2,3]@0 recieved via array \n2. 1@0 recieved via `index` \n3. [1,3]@0 sent via `array`\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/blob/main/reference/2_constructing_data/2_1_setters_deleters/README.md#example---deleting-items-from-arrays",
   "interface": {
     "inputs": {
       "5e2cd3a6-5c09-4da1-8b38-08fa9fc4e955": {
-        "description": "Array to delete item from",
+        "description": "Recieves array to delete item from.",
```
Comment (lvass74) [inline data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json:6]: **the** array

File: data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json [hunk 14]
```diff
@@ -1,22 +1,22 @@
...
         "type": "any[]",
         "name": "array"
       },
       "e576f38e-a208-47e2-a690-abf2d3232f78": {
-        "description": "Identifies item to be deleted",
+        "description": "Recieves index which identifies item to be deleted from the array.",
```
Comment (lvass74) [inline data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json:11]: **the** index

File: data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json [hunk 15]
```diff
@@ -1,22 +1,22 @@
...
         "type": "number",
         "name": "index"
       }
     },
     "outputs": {
       "e685dbbe-2d75-40aa-88dc-a5154e434f3d": {
-        "description": "Array with specified item deleted",
-        "type": "typeof(array)",
+        "description": "Sends array with specified item deleted.",
```
Comment (lvass74) [inline data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json:18]: **the** array

File: data/array/item deleter/internal/37f820aa-4842-43ab-acd4-2c47f999ce08.json [hunk 16]
```diff
@@ -9,7 +9,7 @@
     },
     "outputs": {
       "63ede1b7-2b35-42ec-9116-725f7909d3f6": {
-        "type": "typeof(array & index)[\"array\"]",
+        "type": "(typeof `array & index`)[\"array\"]",
```
Comment (lvass74) [inline data/array/item deleter/internal/37f820aa-4842-43ab-acd4-2c47f999ce08.json:12]: I think it should simply be ```typeof `array` ``` as deletion does not change the type

**Review by lvass74 — CHANGES_REQUESTED** [review]

File: data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json (hunk 14 above)
Comment (danstocker) [reply data/array/item deleter/b12c7233-9990-4473-b3f3-5fced2ac55b8.json:11]: It's fine

File: data/array/item deleter/internal/37f820aa-4842-43ab-acd4-2c47f999ce08.json (hunk 16 above)
Comment (danstocker) [reply data/array/item deleter/internal/37f820aa-4842-43ab-acd4-2c47f999ce08.json:12]: It's an internal, so there is no standalone `array`. It must be extracted from `array & index`.

**Review by danstocker — APPROVED** [review]

**Pull request merged**